src/procstat.h

```diff
@@ -240,6 +240,25 @@ struct procstat_series_u64 {
 	uint64_t aggregated_variance;
 };
 
+
+/**
+ * @brief callback to calculate histogram values
+ */
+typedef void (*percentiles_calculator)(uint32_t *histogram,
+					uint64_t samples_count,
+					struct procstat_percentile_result *result,
+					unsigned result_len);
+
+#define MAX_SUPPORTED_PERCENTILE 20
+struct procstat_histogram_u32 {
+	uint64_t 			  sum;
+	uint64_t 			  count;
+	uint64_t 			  last;
+	int	 			  npercentile;
+	struct procstat_percentile_result percentile[MAX_SUPPORTED_PERCENTILE];
+	uint32_t 			  *histogram;
```

Comment:
@ofir-lb  procstat does not care regards input, it is up to the application to decide units of input.
Input is 32 bit because we keep 1200 buckets per stat, even 32 bits is 4K per statistic .. i did not want it to become 8K. 
Currently data will be displayed in "measurement" format. i.e. if we measure in TSC display will be in TSC. this can be changed later if needed

32 bits is more than enough even for TSC, on 1.7 GHZ it is 1700 clocks in usec 32bit max is over 2 seconds ..